httpx/_decoders.py

```diff
@@ -266,57 +266,23 @@ def __init__(self) -> None:
         self.buffer = ""
 
     def decode(self, text: str) -> typing.List[str]:
-        lines = []
-
-        if text and self.buffer and self.buffer[-1] == "\r":
-            if text.startswith("\n"):
-                # Handle the case where we have an "\r\n" split across
-                # our previous input, and our new chunk.
-                lines.append(self.buffer[:-1] + "\n")
-                self.buffer = ""
-                text = text[1:]
-            else:
-                # Handle the case where we have "\r" at the end of our
-                # previous input.
-                lines.append(self.buffer[:-1] + "\n")
-                self.buffer = ""
-
-        while text:
-            num_chars = len(text)
-            for idx in range(num_chars):
-                char = text[idx]
-                next_char = None if idx + 1 == num_chars else text[idx + 1]
-                if char == "\n":
-                    lines.append(self.buffer + text[: idx + 1])
-                    self.buffer = ""
-                    text = text[idx + 1 :]
-                    break
-                elif char == "\r" and next_char == "\n":
-                    lines.append(self.buffer + text[:idx] + "\n")
-                    self.buffer = ""
-                    text = text[idx + 2 :]
-                    break
-                elif char == "\r" and next_char is not None:
-                    lines.append(self.buffer + text[:idx] + "\n")
-                    self.buffer = ""
-                    text = text[idx + 1 :]
-                    break
-                elif next_char is None:
-                    self.buffer += text
-                    text = ""
-                    break
+        if self.buffer:
+            text = self.buffer + text
+
+        if text.endswith("\r"):
```

Comment:
Tell me more?

Reply:
`splitlines()` will teat a trailing `"\r"` as a newline because it is designed around the assumption that that input is the whole string.. in the case of the line decoder, when a trailing `"\r"` is received, we need to buffer it up in case the next character in the input stream is an `"\n"` (ie. it was part of an interrupted `"\r\n"` sequence)

Reply:
well, if we go with this version then it needs a comment :)

Reply:
Gotcha. Yes a comment would be good. I assume that `"\r\n"` is the *only* two-character newline sequence, right?